File: 2001/3/8-3.cpp
#include <iostream>
#include <vector>
using namespace std;

int main()
{
    vector<int> ary;
    int channel = 1, cross, dim, tmp;
    while (!cin.eof())
    {
        cross = 0;
        ary.clear();
        while (cin >> tmp)
        {
            ary.push_back(tmp);
            if (cin.peek() == '\n')
                break;
        }
        if (ary.empty())
            break;
        dim = ary.size();
        for (int i = 1; i < dim; i++)
            for (int j = 0; j < i; j++)
                if (ary[i] < ary[j])
                    cross++;
        cout << "There are " << cross << " wire crossings in routing channel " << channel++ << endl;
    }
}

File: 2001/3/8-3.py
#!/usr/bin/env python

fin = file("prob3_in.txt", "r")
channel = 1
while 1:
    line = fin.readline()
    if line == "":
        break
    line = line.split()
    ary = []
    for i in line:
        ary.append(int(i))
    cross = 0
    for i in xrange(1, len(ary)):
        for j in xrange(i):
            if ary[i] < ary[j]:
                cross += 1
    print "There are %s wire crossins in routing channel %s." % (cross, channel)
    channel += 1
